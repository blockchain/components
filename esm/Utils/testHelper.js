import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}