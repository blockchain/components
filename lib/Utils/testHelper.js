"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}