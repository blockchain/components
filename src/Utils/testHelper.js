import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
