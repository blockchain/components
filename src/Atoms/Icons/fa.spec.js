import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import { FasArrowLeft } from '.'

describe('fa', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<FasArrowLeft />)

    expect(tree.find(FasArrowLeft)).toMatchSnapshot()
  })

  it('should render correctly with size', () => {
    const tree = mountWithTheme(<FasArrowLeft size="100px" />)

    expect(tree.find(FasArrowLeft)).toMatchSnapshot()
  })

  it('should render correctly with selectable', () => {
    const tree = mountWithTheme(<FasArrowLeft selectable />)

    expect(tree.find(FasArrowLeft)).toMatchSnapshot()
  })
})
