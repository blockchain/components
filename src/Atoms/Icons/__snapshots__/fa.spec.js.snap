// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fa should render correctly 1`] = `
.c0 {
  fill: #545456;
  cursor: default;
}

.c0:hover {
  fill: #545456;
}

<FasArrowLeft
  selectable={false}
  size="44px"
>
  <styled.svg
    height="44px"
    selectable={false}
    viewBox="0 0 448 512"
    width="44px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <svg
      className="c0"
      height="44px"
      viewBox="0 0 448 512"
      width="44px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
      />
    </svg>
  </styled.svg>
</FasArrowLeft>
`;

exports[`fa should render correctly with selectable 1`] = `
.c0 {
  fill: #545456;
  cursor: pointer;
}

.c0:hover {
  fill: #757679;
}

<FasArrowLeft
  selectable={true}
  size="44px"
>
  <styled.svg
    height="44px"
    selectable={true}
    viewBox="0 0 448 512"
    width="44px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <svg
      className="c0"
      height="44px"
      viewBox="0 0 448 512"
      width="44px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
      />
    </svg>
  </styled.svg>
</FasArrowLeft>
`;

exports[`fa should render correctly with size 1`] = `
.c0 {
  fill: #545456;
  cursor: default;
}

.c0:hover {
  fill: #545456;
}

<FasArrowLeft
  selectable={false}
  size="100px"
>
  <styled.svg
    height="100px"
    selectable={false}
    viewBox="0 0 448 512"
    width="100px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <svg
      className="c0"
      height="100px"
      viewBox="0 0 448 512"
      width="100px"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M257.5 445.1l-22.2 22.2c-9.4 9.4-24.6 9.4-33.9 0L7 273c-9.4-9.4-9.4-24.6 0-33.9L201.4 44.7c9.4-9.4 24.6-9.4 33.9 0l22.2 22.2c9.5 9.5 9.3 25-.4 34.3L136.6 216H424c13.3 0 24 10.7 24 24v32c0 13.3-10.7 24-24 24H136.6l120.5 114.8c9.8 9.3 10 24.8.4 34.3z"
      />
    </svg>
  </styled.svg>
</FasArrowLeft>
`;
